contracts/starknet/teleport_router.cairo

```diff
@@ -100,14 +101,16 @@ func auth{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
 
 // constructor(address dai_) {
 @constructor
-func constructor{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(dai: felt) {
+func constructor{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    ward: felt, dai: felt
+) {
     // dai = TokenLike(dai_);
     _dai.write(dai);
 
     let (caller) = get_caller_address();
 
     // wards[msg.sender] = 1;
-    _wards.write(caller, 1);
+    _wards.write(ward, 1);
```

Comment:
Idea here was that contract deployments need to go through an account abstraction. Not sure if it is true.

Reply:
I guess we'll see with Cairo 0.10.1 and this proposal aswell https://community.starknet.io/t/universal-deployer-contract-proposal/1864

Reply:
Right, it is safer to use constructor args right now.